### DISCUSSION TITLE
Welcome to Keyist-Ecommerce Discussions!

### DISCUSSION THREAD

**Discussion by antkaynak** [antkaynak/Keyist-Ecommerce#296] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

You can ask questions about the demo, the structure of the project or actually anything at all about this project!

  🔗 It would be great if you use an already defined issue template if you open an issue. If you need help, here's a [link to the issue templates](https://github.com/antkaynak/Keyist-Ecommerce/tree/master/.github/ISSUE_TEMPLATE).

**Comment by MouadCh** [comment]
I had a  POST http://localhost:8081/oauth/token 403 (Forbidden) ERROR! After configure the params of Database and smtp Protocol.

Chat should i configure to fix this ERROR ? And thnaks in advance

**Reply by antkaynak** [reply]
You might be getting this error because of the clientId and clientSecret fields. Make sure you generated a BCrypt(4) hashed password and filled the oauth_client_details table. I've just [updated the demo sql](https://github.com/antkaynak/Keyist-Ecommerce/blob/master/sql/script.sql) to demonstrate these configurations.

**Reply by MouadCh** [reply]
I update my sql and it worked. But now i got the same error with cart and account/statu... 
- GET http://localhost:8080/api/cart 401 (Unauthorized)
- GET http://localhost:8080/api/account/status 401 (Unauthorized)
- GET http://localhost:8080/api/cart 401 (Unauthorized)
- GET http://localhost:4200/image-url-here 404 (Not Found)

**Reply by antkaynak** [reply]
To be able to use these endpoints you must authenticate first 🙂
This application uses OAuth flow with opaque tokens. Definitely look into auth flow.

Also I see that you did not configure [client's](https://github.com/antkaynak/Keyist-Ecommerce/blob/master/client/src/config/local.ts) configuration. If you want to display images on the client you have to use your own image urls.

**Reply by CharatKakkar** [reply]
https://github.com/antkaynak/Keyist-Ecommerce/blob/master/authorization_server/src/main/java/com/commerce/oauth/config/ServerSecurityConfig.java#L43 Line 43 has a typo. It should be oauth

**Reply by CharatKakkar** [reply]
I can't get this project running. What is the login url?
